User guide: Add test parameters section for SDK python tests (#405)
project-chip/certification-tool: docs/Matter_TH_User_Guide/Matter_TH_User_Guide.adoc

| 38          | 30-Aug-2024  | [Apple] Hilton Lima                 | * Replace images by text examples in 'Project Configuration' section. +
                                                                     * Removed section 'Collect Logs and Submit to TEDS'. +
                                                                     * Added informations about 'qr-code' and 'manual-code' parameters.

> | 39          | 04-Sep-2024  | [Apple] Hilton Lima                 | * Move pixit section. +
>                                                                      * Added 'Test Parameters for SDK Python Tests' section. +
>                                                                      * Change Table of Contents display level.

Reviewer: ```suggestion
| 39          | 04-Sep-2024  | [Apple] Hilton Lima                 | * Moved PIXIT section. +
                                                                     * Added 'Test Parameters for SDK Python Tests' section. +
                                                                     * Changed Table of Contents display level.
```

[source,xml]
----
"dut_config": {
  "discriminator": "3840",
  "setup_code": "20202021",
  "pairing_mode": "ble-wifi",
  "chip_timeout": null,
  "chip_use_paa_certs": false,
  "trace_log": true
}

> ----

Reviewer 2: To maintain the same order as the others pairing modes, I suggest to switch places between the items `a` and `b` of the **Wi-Fi Mode** section.

Reviewer 2: Solved in https://github.com/project-chip/certification-tool/pull/405/commits/d6514cd7fddc91af92a74c9fc3f3ab6d642d7ce1

[source,xml]
----
"thread": {
  "operational_dataset_hex": "0e08000000000001000035060004001fffe00708fd5270f26ee4c02c041064dc641d7195508d7cd17ce22db711420c0402a0f7f8000300000f0102123402081111111122222222030444454d4f051000112233445566778899aabbccddeeff"
}
----
+

> WARNING: _OTBR needs to be configured and running. TH will not start any OTBR dockers._

Reviewer 3: ```suggestion
WARNING: _OTBR needs to be configured and running. TH will not start any OTBR docker containers._
```

[source,xml]
----
"test_parameters": {
  "qr-code": "MT:-24J042C00KA0648G00"
  "manual-code": "34970112332"
}
----
+

> WARNING: This is an invalid configuration. TH will not accept both parameters set at same time.

Reviewer 3: ```suggestion
WARNING: This is an invalid configuration. TH will not accept both parameters set at the same time.
```

    "storage-path": "admin_storage.json",
    "int-arg": "PIXIT.ACE.APPENDPOINT:1 PIXIT.ACE.APPDEVTYPEID:256",
    "string-arg": "PIXIT.ACE.APPCLUSTER:OnOff PIXIT.ACE.APPATTRIBUTE:OnOff"
  }
  ...
}
----

The above example will be used to define the following arguments when running the test:

----
--int-arg PIXIT.ACE.APPENDPOINT:1 PIXIT.ACE.APPDEVTYPEID:256 --string-arg PIXIT.ACE.APPCLUSTER:OnOff PIXIT.ACE.APPATTRIBUTE:OnOff
----
==== Test Parameters for SDK Python Tests
Access the spreadsheet via the https://groups.csa-iot.org/wg/matter-csg/document/36336[Verification Steps Document] and review the information provided. Based on this data, create the parameters set as requested. 

Below are some specific examples assembled from data obtained from the spreadsheet.

. TC-ACE-1.4
+
Sample command to run manually inside docker
+
[source,shell]
----
python3 TC_ACE_1_4.py --discriminator 3840 --passcode 20202021 --commissioning-method on-network --storage-path admin_storage.json  --int-arg PIXIT.ACE.APPENDPOINT:1 PIXIT.ACE.APPDEVTYPEID:256 --string-arg PIXIT.ACE.APPCLUSTER:OnOff PIXIT.ACE.APPATTRIBUTE:OnOff  --paa-trust-store-path /credentials/development/paa-root-certs/
----
+

> Arguments to be used while execution using UI (use product specific values)

Reviewer 4: ```suggestion
Arguments to be used while executing using UI (use product specific values)
```

+
[source,xml]
----
"test_parameters": {
  "int-arg" : "PIXIT.ACE.APPENDPOINT:1 PIXIT.ACE.APPDEVTYPEID:256",
  "string-arg": "PIXIT.ACE.APPCLUSTER:OnOff PIXIT.ACE.APPATTRIBUTE:OnOff"
}
----

. TC-SC-7.1
+
Sample command to run manually inside docker
+
[source,shell]
----
python3 TC_SC_7_1.py --bool-arg post_cert_test:true  --qr-code MT:-<24J0CEK01KA0648G00> --storage-path admin_storage.json --paa-trust-store-path ../../credentials/development/paa-root-certs/
----
+

> Arguments to be used while execution using UI (use product specific values)

Reviewer 4: ```suggestion
Arguments to be used while executing using UI (use product specific values)
```

[source,xml]
----
"test_parameters": {
  "qr-code": "MT:-24J042C00KA0648G00"
  "manual-code": "34970112332"
}
----
+
WARNING: This is an invalid configuration. TH will not accept both parameters set at same time.

On completion of the "network" and the "dut_config" configuration, select the *Update* and then *Create* button to create the Test Project. 

> ==== PIXIT Support

Reviewer: This section is about the same thing as the new `Test Parameters for SDK Python Tests` section (I believe TH only supports PIXIT for python tests, so this section was a way to show how to pass these parameters to those tests). I suggest merging the two sections (maybe just completely substituting this section with the new one would be enough)

Author: Moved pixie section

> PIXIT type parameters must be filled in the *test_parameters* section.

Reviewer 2: Since this section is from the document, please replace with a `ASCIIDOC link` to that section instead.

The above example will be used to define the following arguments when running the test:

[source,shell]
----
--int-arg PIXIT.ACE.APPENDPOINT:1 PIXIT.ACE.APPDEVTYPEID:256 --string-arg PIXIT.ACE.APPCLUSTER:OnOff PIXIT.ACE.APPATTRIBUTE:OnOff
----

====== Test Parameters Examples

Access the spreadsheet via the https://groups.csa-iot.org/wg/matter-csg/document/36336[Verification Steps Document] and review the information provided. Based on this data, create the parameters set as requested. 

Below are some specific examples assembled from data obtained from the spreadsheet.

. TC-ACE-1.4
+
Sample command to run manually inside docker
+
[source,shell]
----

> python3 TC_ACE_1_4.py --discriminator 3840 --passcode 20202021 --commissioning-method on-network --storage-path admin_storage.json  --int-arg PIXIT.ACE.APPENDPOINT:1 PIXIT.ACE.APPDEVTYPEID:256 --string-arg PIXIT.ACE.APPCLUSTER:OnOff PIXIT.ACE.APPATTRIBUTE:OnOff  --paa-trust-store-path /credentials/development/paa-root-certs/

Reviewer 3: @hiltonlima did you test the sample scripts to validate the translation from VS to TH? I'm asking because one of the complains from users is sometimes the mapping between command-line and TH project config values is not direct and we need to gradually identify and correct those cases.

Author: These examples were taken AS IS from the step verification spreadsheet.

I have not run them in my environment.

Author: <img width="1610" alt="image" src="https://github.com/user-attachments/assets/1d85fd57-75b7-42bc-867c-de52ebc1503c">
